src/components/PopupWithForm.js

```diff
@@ -16,6 +18,12 @@ export default class PopupWithForm extends Popup {
     return formInputs;
   }
 
+  setLoading(isLoading, buttonValue) {
+    this._submitButton.textContent = isLoading
+      ? buttonValue
+      : this._submitButtonValue;
+  }
+
   setEventListeners() {
     super.setEventListeners();
     this._popupForm.addEventListener("submit", (evt) => {
```

Comment:
Please, delete `evt.target.reset();` from line 33


You need to **reset** the form only in block `then` in order not to do that immediately after `submit` because when there is an error, the modal will still be opened and the user can try to send the data again to the server without having to type the data into the inputs once more.

Reply:
Thank you for all the advice! 
I deleted the evt.target.reset() from this line but now my forms have the last submission textContent. Would I have to reset the textContent to "" in order to clear out the forms or would I put the evt.target.reset() in a different place?

Reply:
Never mind! I used the FormElement.reset() method in the then block. I think this resolves the issue.